### DISCUSSION TITLE
TL-XTR10890

### DISCUSSION THREAD

**Discussion by ci70** [coolsnowwolf/lede#9775] [Q&A / 快速问答]
如何在其上闪烁openwrt？

**Comment by zhouF96** [comment]
only early version of xtr10890 is supported ( which uboot is not locked ) and you need to flash a third party uboot and mibib instead of the vendor's version. After that you can use it. but currently the qcn90xx do not work.

**Comment by ci70** [comment]
Is it at least possible to root it like Xiaomi?
I need to change the language.

**Reply by zhouF96** [reply]
you can try it yourself but the tplink routers sold in China is different from those sold overseas.
Currently we haven't find a way to hack it. However, you can contact us to buy a nand preflashed with openwrt built for this device and replace the original one.
